<?xml version="1.0"?>
<sdf version="1.4">
	<world name="1grip_1block">
		<include>
			<uri>model://ground_plane</uri>
		</include>
		
		<include>
			<uri>model://sun</uri>
		</include>
		
		<model name="finger">
			<pose>0 1.0 4.0 0 0 0</pose>
			<link name="finger_body">
				<collision name="fbody_collision">
					<geometry>
						<box>
							<size>0.5 0.5 1</size>
						</box>
					</geometry>
				</collision>
				
				<gravity>0</gravity>
				
				<visual name="fbody_visual">
					<geometry>
						<box>
							<size>0.5 0.5 1.2</size>
						</box>
					</geometry>
					<transparency>0.0</transparency>
				</visual>
			</link>
			
			<link name="finger_tip">
				<pose>0 0 -0.75 0 0 0</pose>
				<gravity>0</gravity>

				<visual name="ftip_visual">
					<geometry>
						<box>
							<size>0.4 0.4 0.5</size>
						</box>
					</geometry>
					<transparency>0.5</transparency>
					<material>
						<ambient>1.0 0.0 0.0 0.0</ambient>
					</material>
				</visual>
				<collision name="ftip_collision">
					<geometry>
						<box>
							<size>0.4 0.4 0.5</size>
						</box>
					</geometry>
				</collision>
				
				<sensor name='tip_contact' type='contact'>
					<plugin name="sticky_finger" filename="libsticky_fingers.so">
						<capacity>25</capacity>
					</plugin>
          			<contact>
            			<collision>ftip_collision</collision>
          			</contact>
        		</sensor>
			</link>
			
			<joint name="fixed1" type="prismatic">
 				<parent>finger_body</parent>
 				<child>finger_tip</child>
 				<pose>0 0 0 0 0 0</pose>
 				<axis>
 					<xyz>0 0 1</xyz>
 					<limit>
 						<upper>0</upper>
 						<lower>0</lower>
 					</limit>
				</axis>
  			</joint>
		</model>
		
		<model name="b1">
			<pose> 0 1.0 2.125 0 0 0</pose>
			<link name="b1_body">
				<collision name="b1_collision">
					<geometry>
						<box>
							<size>0.25 0.25 0.25</size>
						</box>
					</geometry>
				</collision>
				<visual name="b1_visual">
					<geometry>
						<box>
							<size>0.25 0.25 0.25</size>
						</box>
					</geometry>
					<material>
						<ambient>0.0 1.0 0.0 0.0</ambient>
					</material>
				</visual>
			</link>
			
			<link name="b2_body">
			
				<pose> 0.5 0 0 0 0 0</pose>
			
				<collision name="b2_collision">
					<geometry>
						<box>
							<size>0.25 0.25 0.25</size>
						</box>
					</geometry>
				</collision>
				<visual name="b2_visual">
					<geometry>
						<box>
							<size>0.25 0.25 0.25</size>
						</box>
					</geometry>
					<material>
						<ambient>0.0 0.0 1.0 0.0</ambient>
					</material>
				</visual>
			</link>
			
			<joint name="fixed2" type="prismatic">
 				<parent>b2_body</parent>
 				<child>b1_body</child>
 				<pose>0 0 0 0 0 0</pose>
 				<axis>
 					<xyz>0 0 1</xyz>
 					<limit>
 						<upper>0.1</upper>
 						<lower>-0.1</lower>
 					</limit>
				</axis>
  			</joint>
		</model>
		
		<model name="table">
			<pose>0 1.0 0.0 0 0 0</pose>
			<include>
			    <inertial>
        			<mass>50</mass>
        		</inertial>
          		<static>false</static>
          		<uri>model://table</uri>
        	</include>
        </model>
	</world>
</sdf>
